rest-api/api/participant_summary_api.py

```diff
@@ -1,16 +1,72 @@
-from api.base_api import BaseApi
-from api_util import PTC_AND_HEALTHPRO
-from app_util import auth_required
-from dao.participant_summary_dao import ParticipantSummaryDao
+import json
 
+import datetime
+from api.base_api import BaseApi, make_sync_results_for_request
+from api_util import PTC_HEALTHPRO_AWARDEE, AWARDEE, DEV_MAIL
+from app_util import auth_required, get_validated_user_info
+from dao.base_dao import json_serial
+from dao.participant_summary_dao import ParticipantSummaryDao
+from flask import request
+from werkzeug.exceptions import Forbidden, BadRequest
+from base64 import urlsafe_b64encode
 
 class ParticipantSummaryApi(BaseApi):
   def __init__(self):
     super(ParticipantSummaryApi, self).__init__(ParticipantSummaryDao())
 
-  @auth_required(PTC_AND_HEALTHPRO)
+  @auth_required(PTC_HEALTHPRO_AWARDEE)
   def get(self, p_id=None):
+    auth_awardee = None
+    user_email, user_info = get_validated_user_info()
+    if AWARDEE in user_info['roles']:
+      if user_email == DEV_MAIL:
+        auth_awardee = request.args.get('awardee')
+      else:
+        try:
+          if user_info['awardee']:
+            auth_awardee = user_info['awardee']
+          elif user_info['organization']:
+            auth_awardee = user_info['organization']
+          else:
+            auth_awardee = user_info['site']
+        except KeyError:
+          raise BadRequest("Must supply awardee for request.")
+
+    # data only for user_awardee, assert that query has same awardee
     if p_id:
+      if auth_awardee and user_email != DEV_MAIL:
+        raise Forbidden
       return super(ParticipantSummaryApi, self).get(p_id)
     else:
+      if auth_awardee:
+        # make sure request has awardee
+        requested_awardee = request.args.get('awardee')
+        if requested_awardee != auth_awardee:
+          raise Forbidden
       return super(ParticipantSummaryApi, self)._query('participantId')
+
+  def _make_query(self):
```

Comment:
I don't think this will work... with the API as-is, the client isn't going to be passing up lastModified as a request param (which would cause field_filters to be populated)... it will instead pass up a pagination token, which is used directly to filter the SQLAlchemy query here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/base_dao.py#L209

So I think what we need to do is override ParticipantSummaryDao._decode_token() -- that's where the last modified timestamp passed up in the pagination token will need to be tweaked.

If somebody *does* explicitly specify lastModified as a request param, we should use the provided request param value as-is without modification.

Sorry if I got this wrong last week. This code is confusing. :)

Reply:
done